### ISSUE TITLE
Let execute some commands before moving files

### ISSUE THREAD

**Issue by kwiat1990** [appleboy/scp-action#46]
Hi,

I ran into some issues with current action. On my server I have files (dot files like `.htacess` and `.env`), which need to stay untouched, while other files need to be replaced. As some of the files to replace contain hash in their names, there are multiple copies of the same file on the server (after every new deployment a new one comes). So ideally I would like delete all files in the target directory but dot files and then move rest of the files from my repository.

**Comment by carolusjoseph** [comment]
as i know in documentation there is can ignore list

**Comment by kevinmamaqi** [comment]
Is it possible to erase destination folder before deployment?

**Comment by carolusjoseph** [comment]
> Is it possible to erase destination folder before deployment?

use another plugin first ([](https://github.com/marketplace/actions/ssh-remote-commands))

**Comment by kwiat1990** [comment]
> as i know in documentation there is can ignore list

Yeah, it is mention in docs but it won't work in this case as `rm` command takes only a boolean, so it either removes the entire target directory or does nothing. No options can be passed.

**Comment by appleboy** [comment]
@kwiat1990 OK, maybe I can add new parameter to setup the ignore list and replace the `rm` with `find ... -exec` command like as below:

```sh
find . ! -name 'file.txt' -type f -exec rm -f {} +
```

**Comment by kwiat1990** [comment]
@appleboy hi, thanks for you feedback. It seems to be a good idea. In my case I need to preserve multiple files, so from my perspective it would be important that the command let me do that.

> ```shell
> find . ! -name 'file.txt' -type f -exec rm -f {} +
> ```

**Comment by billyjov** [comment]
@appleboy Any news on this use case?
